From 570d235632c786da6f2bd6836d8589276b9f7cb9 Mon Sep 17 00:00:00 2001
From: Andy Doan <andy@foundries.io>
Date: Tue, 17 Sep 2019 10:05:39 -0500
Subject: [PATCH 01/12] SQL: Add a new column to store "custom" data for Target

As more people take advantage of the custom dictionary value for
a Target, we'll need to keep track of this so things like package
managers can reason about things correctly.

Signed-off-by: Andy Doan <andy@foundries.io>
---
 config/sql/migration/migrate.21.sql |  9 +++++++++
 config/sql/rollback/rollback.21.sql | 13 +++++++++++++
 config/sql/schema.sql               |  4 ++--
 3 files changed, 24 insertions(+), 2 deletions(-)
 create mode 100644 config/sql/migration/migrate.21.sql
 create mode 100644 config/sql/rollback/rollback.21.sql

diff --git a/config/sql/migration/migrate.21.sql b/config/sql/migration/migrate.21.sql
new file mode 100644
index 00000000..95a5a751
--- /dev/null
+++ b/config/sql/migration/migrate.21.sql
@@ -0,0 +1,9 @@
+-- Don't modify this! Create a new migration instead--see docs/schema-migrations.adoc
+SAVEPOINT MIGRATION;
+
+ALTER TABLE installed_versions ADD COLUMN custom_meta TEXT NOT NULL DEFAULT "";
+
+DELETE FROM version;
+INSERT INTO version VALUES(21);
+
+RELEASE MIGRATION;
diff --git a/config/sql/rollback/rollback.21.sql b/config/sql/rollback/rollback.21.sql
new file mode 100644
index 00000000..01ca624a
--- /dev/null
+++ b/config/sql/rollback/rollback.21.sql
@@ -0,0 +1,13 @@
+-- Don't modify this! Create a new migration instead--see docs/schema-migrations.adoc
+SAVEPOINT ROLLBACK_MIGRATION;
+
+CREATE TABLE installed_versions_migrate(id INTEGER PRIMARY KEY, ecu_serial TEXT NOT NULL, sha256 TEXT NOT NULL, name TEXT NOT NULL, hashes TEXT NOT NULL, length INTEGER NOT NULL DEFAULT 0, correlation_id TEXT NOT NULL DEFAULT '', is_current INTEGER NOT NULL CHECK (is_current IN (0,1)) DEFAULT 0, is_pending INTEGER NOT NULL CHECK (is_pending IN (0,1)) DEFAULT 0, was_installed INTEGER NOT NULL CHECK (was_installed IN (0,1)) DEFAULT 0);
+INSERT INTO installed_versions_migrate(ecu_serial, sha256, name, hashes, length, correlation_id, is_current, is_pending, was_installed) SELECT installed_versions.ecu_serial, installed_versions.sha256, installed_versions.name, installed_versions.hashes, installed_versions.length, installed_versions.correlation_id, installed_versions.is_current, installed_versions.is_pending, installed_versions.was_installed FROM installed_versions;
+
+DROP TABLE installed_versions;
+ALTER TABLE installed_versions_migrate RENAME TO installed_versions;
+
+DELETE FROM version;
+INSERT INTO version VALUES(20);
+
+RELEASE ROLLBACK_MIGRATION;
diff --git a/config/sql/schema.sql b/config/sql/schema.sql
index 17aa216e..94e025ba 100644
--- a/config/sql/schema.sql
+++ b/config/sql/schema.sql
@@ -1,9 +1,9 @@
 CREATE TABLE version(version INTEGER);
-INSERT INTO version(rowid,version) VALUES(1,20);
+INSERT INTO version(rowid,version) VALUES(1,21);
 CREATE TABLE device_info(unique_mark INTEGER PRIMARY KEY CHECK (unique_mark = 0), device_id TEXT, is_registered INTEGER NOT NULL DEFAULT 0 CHECK (is_registered IN (0,1)));
 CREATE TABLE ecu_serials(id INTEGER PRIMARY KEY, serial TEXT UNIQUE, hardware_id TEXT NOT NULL, is_primary INTEGER NOT NULL DEFAULT 0 CHECK (is_primary IN (0,1)));
 CREATE TABLE misconfigured_ecus(serial TEXT UNIQUE, hardware_id TEXT NOT NULL, state INTEGER NOT NULL CHECK (state IN (0,1)));
-CREATE TABLE installed_versions(id INTEGER PRIMARY KEY, ecu_serial TEXT NOT NULL, sha256 TEXT NOT NULL, name TEXT NOT NULL, hashes TEXT NOT NULL, length INTEGER NOT NULL DEFAULT 0, correlation_id TEXT NOT NULL DEFAULT '', is_current INTEGER NOT NULL CHECK (is_current IN (0,1)) DEFAULT 0, is_pending INTEGER NOT NULL CHECK (is_pending IN (0,1)) DEFAULT 0, was_installed INTEGER NOT NULL CHECK (was_installed IN (0,1)) DEFAULT 0);
+CREATE TABLE installed_versions(id INTEGER PRIMARY KEY, ecu_serial TEXT NOT NULL, sha256 TEXT NOT NULL, name TEXT NOT NULL, hashes TEXT NOT NULL, length INTEGER NOT NULL DEFAULT 0, correlation_id TEXT NOT NULL DEFAULT '', is_current INTEGER NOT NULL CHECK (is_current IN (0,1)) DEFAULT 0, is_pending INTEGER NOT NULL CHECK (is_pending IN (0,1)) DEFAULT 0, was_installed INTEGER NOT NULL CHECK (was_installed IN (0,1)) DEFAULT 0, custom_meta TEXT NOT NULL DEFAULT "");
 CREATE TABLE primary_keys(unique_mark INTEGER PRIMARY KEY CHECK (unique_mark = 0), private TEXT, public TEXT);
 CREATE TABLE tls_creds(ca_cert BLOB, ca_cert_format TEXT,
                        client_cert BLOB, client_cert_format TEXT,
-- 
2.23.0

